magento/pwa-studio
# [PWA-1342] Upgrade workbox to v6

## Description

Story
As a developer, I would like to upgrade workbox to the latest version so that we can take advantage of improvements to how precache assets requests are fired from the ServiceWorker. Switching from v5 Promise.all flood to v6 one request at a time means that the main app does not have to compete for bandwidth with these precache requests that aren't even needed for initial render. High level testing has shown dramatic improvements in simulated 3G mobile Lighthouse tests.

The workbox upgrade also requires an upgrade to webpack to at least version 4.40.0, but I have found no issue going to the latest 4.46.0.

Acceptance Criteria
- workbox-* dependencies upgraded to 6.0.2 (or latest)
- webpack upgraded to 4.46.0
- Lighthouse Mobile performance metrics should improve

## Related Issue





* [[PWA-1342](https://jira.corp.magento.com/browse/PWA-1342)] Upgrade workbox to v6

## Acceptance



### Verification Stakeholders

### Specification



### Verification Steps


1. `yarn watch:venia` with `DEV_SERVER_SERVICE_WORKER_ENABLED=true`
2. Verify no errors are in the console related to ServiceWorker registration
3. Do a full build (`yarn build && yarn stage:venia`)
4. Verify no errors during build compilation and app is served
5. Verify the upgrade worked by observing precache asset requests in the Performance tab of DevTools (requests should fire one at a time, instead of a big flood). Precache assets should follow the `<number>.<hash>.js` pattern.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
